src/Atmos/Model/AtmosModel.jl

```diff
@@ -896,4 +898,138 @@ function numerical_flux_first_order!(
         fluxᵀn.tracers.ρχ -= ((w1 + w2) * χ̃ + wt) / 2
     end
 end
+
+function numerical_flux_first_order!(
+    ::HLLCNumericalFlux,
+    balance_law::AtmosModel,
+    fluxᵀn::Vars{S},
+    normal_vector::SVector,
+    state_prognostic⁻::Vars{S},
+    state_auxiliary⁻::Vars{A},
+    state_prognostic⁺::Vars{S},
+    state_auxiliary⁺::Vars{A},
+    t,
+    direction,
+) where {S, A}
+    FT = eltype(fluxᵀn)
+    num_state_prognostic = number_states(balance_law, Prognostic())
+    param_set = balance_law.param_set
+
+    flux⁻ = similar(parent(fluxᵀn), Size(3, num_state_prognostic))
+    fill!(flux⁻, -zero(FT))
+    flux_first_order!(
+        balance_law,
+        Grad{S}(flux⁻),
+        state_prognostic⁻,
+        state_auxiliary⁻,
+        t,
+        direction,
+    )
+    fluxᵀn⁻ = flux⁻' * normal_vector
+
+    flux⁺ = similar(flux⁻)
+    fill!(flux⁺, -zero(FT))
+    flux_first_order!(
+        balance_law,
+        Grad{S}(flux⁺),
+        state_prognostic⁺,
+        state_auxiliary⁺,
+        t,
+        direction,
+    )
+    fluxᵀn⁺ = flux⁺' * normal_vector
+
+    ρ⁻ = state_prognostic⁻.ρ
+    ρu⁻ = state_prognostic⁻.ρu
+    ρe⁻ = state_prognostic⁻.ρe
+    ts⁻ = thermo_state(
```

Comment:
Is this expected to work at the boundaries? At the boundary, `state_auxiliary⁻` (specifically variables that are a function of the state) is not necessarily in sync with the state assigned in `boundary_state!`, in which case, you'll get different results by creating a new thermodynamic state:
```julia
ts⁻ = PhaseEquil(
        atmos.param_set,
        internal_energy(atmos, state, aux),
        state⁻.ρ,
        state⁻.moisture.ρq_tot / state.ρ,
        moist.maxiter,
        atmos.moisture.tolerance,
    )
```
The problem with this is that it will call saturation adjustment.